crates/ty_ide/src/symbols.rs

```diff
@@ -622,4 +1067,837 @@ mod tests {
+from collections import defaultdict as defaultdict
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_assignment() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = ['defaultdict']
+").exports(),
+            @"defaultdict :: Variable",
+        );
+
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = ('defaultdict',)
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_annotated_assignment() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__: list[str] = ['defaultdict']
+").exports(),
+            @"defaultdict :: Variable",
+        );
+
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__: tuple[str, ...] = ('defaultdict',)
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_augmented_assignment() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = []
+__all__ += ['defaultdict']
+").exports(),
+            @"defaultdict :: Variable",
+        );
+
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = []
+__all__ += ('defaultdict',)
+").exports(),
+            @"defaultdict :: Variable",
+        );
+
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = []
+__all__ += {'defaultdict'}
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_invalid_augmented_assignment() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ += ['defaultdict']
+").exports(),
+            @"",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_extend() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = []
+__all__.extend(['defaultdict'])
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_invalid_extend() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__.extend(['defaultdict'])
+").exports(),
+            @r"",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_append() {
+        insta::assert_snapshot!(
+            public_test("\
+from collections import defaultdict
+__all__ = []
+__all__.append('defaultdict')
+").exports(),
+            @"defaultdict :: Variable",
+        );
+    }
+
+    #[test]
+    fn exports_from_import_all_reexport_remove() {
```

Comment:
I don't see any tests for the fact that a subsequent non-augmented assignment (or import) of `__all__` clears previous names. (Or any fixme test for the fact that we'll do this even if they are conditional, e.g. in

```py
if whatever:
    __all__ = [...]
else:
    __all__ = [...]
```

we will always take the `else` version of `__all__`.

Reply:
Nice catch. Tests added!